### PR TITLE
Initial setup

### PR DESCRIPTION
**Pull request by phackstock**
Initial setup with the necessary changes plus the demonstration from last Wednesday

### CODE REVIEW

File: 2021-05-26-type-hints/test_pandas.py [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+    return pd.read_csv(file)
+
+
+if __name__ == "__main__":
+    df1 = read_data("sample_data.csv")
+    df2 = read_data(
+        Path("/home/hackstock/Documents/code/MESSAGEix-weekly") / "sample_data.csv"
+    )
+    assert_frame_equal(df1, df2)
+    print(df1.head(3))
```
Comment (danielhuppmann) [inline 2021-05-26-type-hints/test_pandas.py]: Not quite sure why you would want to print here

File: 2021-05-26-type-hints/test_pandas.py [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+
+
+def read_data(file: Union[str, Path]) -> pd.DataFrame:
+    return pd.read_csv(file)
+
+
+if __name__ == "__main__":
+    df1 = read_data("sample_data.csv")
+    df2 = read_data(
+        Path("/home/hackstock/Documents/code/MESSAGEix-weekly") / "sample_data.csv"
```
Comment (danielhuppmann) [inline 2021-05-26-type-hints/test_pandas.py]: if possible, avoid using hard-coded file paths

File: README.md [hunk 3]
```diff
@@ -1,57 +1,18 @@
...
 ## Overview
 
-Template repository for creating python packages and Sphinx-based documentation pages in line with the IIASA design guidelines
+This repository contains a collection of small code snippets presented during the MESSAGEix
+weekly meetings.
 
-## Configuration
+The idea is that each presentation is centered around a jupyter notebook. In case a single
+notebook is sufficient, the suggested naming convention is
+`YYYY-MM-DD-name-of-the-topic.ipynb`. If you have multiple files, please create a directory
```
Comment (danielhuppmann) [inline README.md:15]: I would rather place all tutorial notebooks in a folder `tutorials` with a descriptive name, and have a list of all notebooks here in the Readme following the style

```
 - topic of the notebook (<name of presenter>, YYY-MM-DD) [link-to-notebook]:
   short description, keywords, ...
```

File: 2021-05-26-type-hints/test_pandas.py [hunk 4]
```diff
@@ -0,0 +1,18 @@
+import pandas as pd
```
Comment (danielhuppmann) [inline 2021-05-26-type-hints/test_pandas.py]: Not clear to me why this file is helpful - why not just have everything in the notebook?

**Review by danielhuppmann — COMMENTED** [review]
Thanks @phackstock for setting up the repo for the code-and-tell section.

On the structure, I'd ask @khaeru to also chime in (see my comment for details, I suggest to have an overview list in the readme and then a folder `tutorials` with Jupyter notebooks (or other file types in the future).

Also, please explain why the test is helpful, and replace the csv file by a smaller example - no need to have 5MB here.

**Comment by khaeru** [comment]
My thoughts were:
- This is doesn't need to be a package. Thus:
  - No docs.
  - No tests.
  - No CI.
- Thus, simplest to put everything at the **very top level** of the repo. Go to the URL → see the list of demos.
- Use names with the date and who gave the demo, e.g. `2021-05-31-kishimoto.ipynb`.
- If multiple files for a given date, then a directory `2021-05-31-kishimoto/` with files `code.py` and `data.csv`.
- List only titles in the README:

  ```
  - 2021-05-31 (@phackstock): Type hinting
  ```

Let's remember the original purpose that was imagined: a place to collect these quick, informal demos shared within our team during our weekly meetings. Every additional requirement increases the cost to our colleagues to add to the repo; the practical result will be fewer people contributing less frequently.

If we want to do other things, like collect well-polished teaching materials, we already have places to do that.

**Comment by khaeru** [comment]
> * This is doesn't need to be a package.

Put another way: this is not really an intended use-case for the python-stub template. The first question should usually be: "Do I need to use this template?" and the answer here would have been "No." It might be nice to add that there: a section/prompt like "When to use this template".

**Comment by danielhuppmann** [comment]
Thanks for the comments, @khaeru.

Agree with the suggested structure, though it might be more helpful to have a file/folder name like `<date>-<title>[-<author>]`. The content is usually more interesting than the person who presented a code-and-tell.

And I disagree that the docs are not in the scope of the template repo - I find it super-useful that the tutorial notebooks for pyam (see [here](https://pyam-iamc.readthedocs.io/en/latest/tutorials/pyam_first_steps.html)) are rendered directly on ReadTheDocs, in particular because GitHub notebook rendering sometimes fails with larger files. in my opinion, something to add add here as a later improvement (and maybe also at some point for MESSAGEix).

**Comment by khaeru** [comment]
> I disagree that the docs are not in the scope of the template repo

I didn't suggest otherwise. I meant that the things that the template repo is designed to help with are not requirements in this particular use-case, thus there wasn't a need to use it.

That includes docs: "Here is a code-and-tell notebook; and here are docs *about* the notebook" is already an unnecessary distinction. Like the one given today, the notebooks should be self-contained.

**Comment by danielhuppmann** [comment]
Sorry, I meant of course this repo. And I don't mean adding docs *about this repo* but using sphinx to render the notebooks on a show-and-tell docs page. It's an extra minute when adding a new code-and-tell notebook to this repo to include it in the list of rendered notebooks and making sure that dependencies are included, but it makes it easier and faster to quickly browse through a dozen notebooks to find the one item that you are looking for...

**Comment by phackstock** [comment]
Cleaned up the repo and streamlined the layout to now only having top level jupyter notebooks.

**Review by danielhuppmann — APPROVED** [review]
Lgtm, @phackstock. In spite of the general reservations by @khaeru, let's proceed and collect a few notebooks here and see how this develops.

**Pull request merged**